interfaces/cython/SConscript

```diff
@@ -180,21 +180,29 @@ if localenv['python2_package'] == 'full':
 
     # Use 3to2 to convert examples from Python 3 syntax
     if env['python_convert_examples']:
-        def convert_example(target, source, env):
-            shutil.copyfile(source[0].abspath, target[0].abspath)
-            subprocess.call(env['threetotwo_cmd'] + ['--no-diff', '-n', '-w','-x', 'str',
-                         '-f', 'all', '-f', 'printfunction', '-x', 'print',
-                         '-x', 'open', target[0].abspath])
-        for subdir in subdirs('cantera/examples'):
-            dirpath = pjoin('cantera', 'examples', subdir)
-            for filename in os.listdir(dirpath):
-                if not filename.endswith('.py'):
-                    continue
-                targetdir = '../../build/python2/cantera/examples'
-                a = build(py2env.Command(pjoin(targetdir, subdir, filename),
-                                         pjoin(dirpath, filename),
-                                         convert_example))
-                py2env.Depends(mod, a)
+        # def convert_examples(target, source, env):
+        #     subprocess.call([env['python2_cmd'], '-c', "from lib3to2.main import main; "
+        #                      "main('lib3to2.fixes', ['--no-diff', '-n', '-w','-x', 'str', '-f',"
+        #                      "'all', '-f', 'printfunction', '-x', 'print', '-x', 'open', {}]".format(target[0].abspath)])
+        a = build(py2env.Command('#build/python2/cantera/examples', 'cantera/examples',
+                                 Copy("$TARGET", "$SOURCE")))
+        from textwrap import dedent
+        convert_script = dedent("""\
+        $python2_cmd_esc -c "from lib3to2.main import main; main('lib3to2.fixes', ['--no-diff', '-n', '-w','-x', 'str', '-f', 'all', '-f', 'printfunction', '-x', 'print', '-x', 'open', '../../build/python2/cantera/examples'])"
+        """)
+        b = build(py2env.Command([], '#build/python2/cantera/examples', convert_script))
+        py2env.Depends(b, a)
+        py2env.Depends(mod, b)
```

Comment:
@speth I'm really not sure what's going on here. I'm trying to run the converter before the module is built, and after the example files are copied to the right place. Instead, the setup script is copying the example files and isn't converting them. I don't see how what I'm doing here is different from how it used to be in terms of the dependency, but clearly something is different. Any advice would be appreciated.

Reply:
Is it because the `Depends` is on a folder, rather than a set of files?

Reply:
I think the issue is that in your new way, `b` is an empty list (because you specify an empty list as the `target` for the `Command`), so there's no state for SCons to base the dependency on. I think what you actually want here is `env.AddPostAction`, which will run after `a`, but only if the `a` action has to run in the first place.